zerver/tests/test_upload.py

```diff
@@ -1574,9 +1576,12 @@ def test_file_upload_authed(self) -> None:
         self.assertEqual(base, uri[:len(base)])
 
         response = self.client_get(uri)
-        redirect_url = response['Location']
+        self.assertEqual(response.status_code, 302)
 
-        self.assertEqual(b"zulip!", urllib.request.urlopen(redirect_url).read().strip())
```

Comment:
Here's the `403` error I mentioned in chat. I couldn't find a better way to get keys from reading the docs, other than iterating through them.